mesonbuild/backend/backends.py

```diff
@@ -329,7 +329,7 @@ def generate_basic_compiler_args(self, target, compiler, no_warn_args=False):
         commands += compiler.get_always_args()
         if no_warn_args:
             commands += compiler.get_no_warn_args()
-        else:
+        elif self.environment.coredata.get_builtin_option('buildtype') != 'plain':
```

Comment:
This is resolving the problem that 'plain' isn't actually 'plain' in Meson but adds some warning flags unconditionally.
On distributions, if we e.g. want to enable only specific warnings (e.g. implicit-function-declaration) and make them fatal, this behavior would be an issue.

It's a separate bug though and would be in a separate PR if Github would let me do that.